### ISSUE TITLE
Rule proposal: `prefer-filter-boolean`

### ISSUE THREAD

**Issue by B2o5T** [sindresorhus/eslint-plugin-unicorn#1564]
It's more readable that returned values will be only `truthy`

## Fail

```js
list.filter(v => v)
list.filter(v => !!v)
list.filter(v => Boolean(v))
```

## Pass

```js
list.filter(Boolean)
list.filter(v => !v)
list.filter(v => v === true)
list.filter(v => v > 0)
```

**Comment by sindresorhus** [comment]
It's not clear to me why `list.filter(Boolean)` should be preferred over `list.filter(v => Boolean(v))`. And it would also conflict with one of our rules: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md

**Comment by fisker** [comment]
> And it would also conflict with one of our rules

Actually. no. `Boolean` is ignored.

**Comment by B2o5T** [comment]
@sindresorhus in your case `list.filter(Boolean)` is preferred because less code is written, but the main purpose of `.filter(Boolean)` is - we no need to think about callback logic of `filter` method, we know that returned values will be truthy.

**Comment by B2o5T** [comment]
found similar rule from `putout`, but currently handle only `v => v` case https://github.com/coderaiser/putout/tree/master/packages/plugin-apply-filter-boolean

**Comment by fisker** [comment]
I'm 👍 for this rule.

**Comment by B2o5T** [comment]
we can make same as `putout` package <https://github.com/coderaiser/putout/releases/tag/v21.6.0>
So I propose a rule with the name `no-useless-functions` that will catch failed test cases for `filter`, `find`, `findIndex`, `some`, `every` array methods

**Comment by sindresorhus** [comment]
This is now accepted.
